### PR TITLE
[Tests] add failing tests for proxies of functions

### PR DESCRIPTION
**Pull request by CrocoDillon**
Title says it all 😄 

Expected a proxy of a callable (in this case a mere function) to pass, but it doesn't.

Of course this shouldn't be merged before the currently failing test passes!

### CODE REVIEW

**Comment by ljharb** [comment]
Thanks, I'll take a look.

**Comment by ljharb** [comment]
Looks like `Function.prototype.toString.call(new Proxy(() => {}, {}))` throws, because it's brand-checking the proxy and it knows it's not a function. I'm looking into it.

**Comment by CrocoDillon** [comment]
Awesome, thanks for looking into this!

I was running into the issue using Enzyme (which depends on `function.prototype.name`, which depends on `is-callable`) because for testing we mock some React components using proxies.

Sadly I don't have any clue how to solve this otherwise I would be of more help. Hope you can figure it out, good luck! 😄

**Comment by ljharb** [comment]
I'm surprised you're using proxies to mock components; i'd think it's much easier to mock out the methods you need to test - that sounds like a really brittle way to test them.

**Comment by CrocoDillon** [comment]
Let's just say it's a long story 😉

**Comment by ljharb** [comment]
k, I've tried this again, and there's really no way to reliably cover this.

In other words, a Proxy for a function can pass via `typeof x === 'function'`, but in order to robustly *fail* for things that are typeof function but not actually functions (like regexes in some engines), I have to use `Function.prototype.toString.call`, which fails on a Proxy because internal slots aren't forwarded.

In other words, I think this can't be fixed.

**Comment by CrocoDillon** [comment]
Ah well, that kinda sucks 😅 

Thanks for trying though! Maybe someone will find this issue and comes up with a reliable test. Meanwhile I guess you can close this. Thanks again!

**Comment by sirian** [comment]
More false positive scenarios #30

**Comment by CrocoDillon** [comment]
Cool 😄 👍
